packages/vite/src/node/http.ts

```diff
@@ -205,3 +194,34 @@ export function setClientErrorHandler(
     socket.end(`HTTP/1.1 ${msg}\r\n\r\n`)
   })
 }
+
+async function findAvailablePort(
+  port: number,
+  logger: Logger,
+  host?: string,
+  strictPort?: boolean,
+): Promise<number> {
+  return new Promise((resolve, reject) => {
+    if (port > 65535) {
+      reject(new Error('Port not found'))
+    }
+    const server = net.connect(port, host)
+    server.on('connect', async () => {
+      if (strictPort) {
+        reject(new Error(`Port ${port} is already in use`))
+      } else {
+        logger.info(`Port ${port} is in use, trying another one...`)
+        const _port = await findAvailablePort(++port, logger, host, strictPort)
+        resolve(_port)
+      }
```

Comment:
```suggestion
      }
      if (!server.closed) {
        server.end()
      }
```
I guess we should call `server.end()`.